production/grafanacloud-install.ps1

```diff
@@ -1,41 +1,51 @@
-# Script to install Grafana agen for Windows
-param ($GCLOUD_STACK_ID, $GCLOUD_API_KEY, $GCLOUD_API_URL)
+#Requires -RunAsAdministrator
+
+# Script to install Grafana agent for Windows
+param ($GCLOUD_HOSTED_METRICS_URL, $GCLOUD_HOSTED_METRICS_ID, $GCLOUD_SCRAPE_INTERVAL, $GCLOUD_HOSTED_LOGS_URL, $GCLOUD_HOSTED_LOGS_ID, $GCLOUD_RW_API_KEY)
 
 Write-Host "Setting up Grafana agent"
 
-if ( -Not [bool](([System.Security.Principal.WindowsIdentity]::GetCurrent()).groups -match "S-1-5-32-544") ) {
-  Write-Host "ERROR: The script needs to be run with Administrator privileges"
-  exit
+# Check if required parameters are present
+if ($GCLOUD_HOSTED_METRICS_URL -eq "") {
+	Write-Host "ERROR: Required argument GCLOUD_HOSTED_METRICS_URL missing"
+	exit
 }
 
-# Check if required parameters are present
-if ($GCLOUD_STACK_ID -eq "") {
-  Write-Host "ERROR: Required argument GCLOUD_STACK_ID missing"
-  exit
+if ($GCLOUD_HOSTED_METRICS_ID -eq "") {
+	Write-Host "ERROR: Required argument GCLOUD_HOSTED_METRICS_ID missing"
+	exit
+}
+
+if ($GCLOUD_SCRAPE_INTERVAL -eq "") {
+	Write-Host "ERROR: Required argument GCLOUD_SCRAPE_INTERVAL missing"
+	exit
 }
 
-if ($GCLOUD_API_KEY -eq "") {
-  Write-Host "ERROR:  Required argument GCLOUD_API_KEY missing"
-  exit
+if ($GCLOUD_HOSTED_LOGS_URL -eq "") {
+	Write-Host "ERROR: Required argument GCLOUD_HOSTED_LOGS_URL missing"
```

Comment:
You might want to do a `throw` instead. Please see [this](https://github.com/grafana/agent/pull/2899#discussion_r1097646324) comment.

Reply:
I instead changed it to verify that these parameters has been passed "the PowerShell way" (using `[Parameter(Mandatory = $true)`).

However, I now see this on `main`: https://github.com/grafana/agent/commit/11f90108fad74be7df6a60c81774994e73f04362

Not sure what this new mechanism is since the instructions still mentions downloading and running this script: https://storage.googleapis.com/cloud-onboarding/agent/scripts/grafanacloud-install.ps1

Reply:
And I would argue that this script should still be updated with the changes in this PR somehow...